src/App.tsx

```diff
@@ -1,223 +1,193 @@
+
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filterType, setFilterType] = useState<FilterType>('All');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [todoIdForDeleting, setTodoIdForDeleting] = useState<number[]>([]);
+
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  useEffect(() => {
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => setErrorMessage('Something went wrong'));
+    }
+  }, []);
+
+  const activeTodosAmount = useMemo(() => (
+    todos.filter(todo => !todo.completed).length
+  ), [todos]);
+
+  const isTodoCompleted = useMemo(() => (
+    todos.some(todo => todo.completed)
+  ), [todos]);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filterType) {
+      case 'Completed':
+        return todo.completed;
+      case 'Active':
+        return !todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  const addTodoOn = async () => {
+    setIsAdding(true);
+
+    if (user) {
+      try {
+        setTempTodo({
+          id: 0,
+          userId: user?.id,
+          title: title.trim(),
+          completed: false,
+        });
+
+        const newTodo = await addTodo({
+          userId: user?.id,
+          title: title.trim(),
+          completed: false,
+        });
+
+        setTitle('');
+
+        setTodos(current => [
+          ...current,
+          {
+            id: newTodo.id,
+            userId: newTodo.userId,
+            title: newTodo.title,
+            completed: newTodo.completed,
+          },
+        ]);
+      } catch (error) {
+        // eslint-disable-next-line max-len
+        setErrorMessage('Something went wrong.Unable to add a todo');
+      } finally {
+        setIsAdding(false);
+        setTempTodo(null);
+      }
+    }
+  };
+
+  const handleSubmit = useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      if (!title.trim()) {
+        setErrorMessage('Title can\'t be empty');
+
+        return;
+      }
+
+      addTodoOn();
+    }, [title],
+  );
+
+  const removeTodo = useCallback(async (todoId: number) => {
+    try {
+      setTodoIdForDeleting(currentTodoIds => (
+        [...currentTodoIds, todoId]
+      ));
+
+      await deleteTodo(todoId);
+
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+    } catch (error) {
+      setErrorMessage('Something went wrong.Unable to delete a todo');
```

Comment:
It is a good practice to use ENUM for error messages
```suggestion
      setErrorMessage(TodoErrors.UnableToDelete));
```